mixins/labelled/labelled-mixin.js

```diff
@@ -96,20 +97,23 @@ export const LabelledMixin = superclass => class extends superclass {
 		this.labelRequired = true;
 		this._labelElem = null;
 		this._missingLabelErrorHasBeenThrown = false;
-		this._validatingLabelTimeout = null;
-	}
-
-	firstUpdated(changedProperties) {
-		super.firstUpdated(changedProperties);
-		this._validateLabel(); // need to check this even if "label" isn't updated in case it's never set
+		this.addRequiredProperty('label', {
+			message: defaultMessage => {
+				if (!this.labelledBy) return defaultMessage;
+				return `LabelledMixin: "${this.tagName.toLowerCase()}" is labelled-by="${this.labelledBy}", but its label is empty`;
+			},
+			validator: hasValue => {
+				if (!this.labelRequired || hasValue) return true;
+				if (!this.labelledBy) return false;
+				return this._labelElem !== null;
+			}
+		});
```

Comment:
This is probably the most complex use-case imaginable with a custom validator and custom message (sometimes).